firebase-database/src/main/java/com/google/firebase/database/DatabaseTransaction.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.database;
+
+import androidx.annotation.NonNull;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.database.annotations.Nullable;
+import java.util.concurrent.ExecutionException;
+
+public class DatabaseTransaction {
+
+  public static class DatabaseTransactionContext {
+
+    private final com.google.firebase.database.core.Transaction transaction;
+    private final FirebaseDatabase database;
+
+    DatabaseTransactionContext(
+        com.google.firebase.database.core.Transaction transaction, FirebaseDatabase database) {
+      this.transaction = transaction;
+      this.database = database;
+    }
+
+    @Nullable
+    public DataSnapshot get(@NonNull Query query) throws FirebaseDatabaseException {
+      try {
+        return Tasks.await(transaction.get(query));
```

Comment:
For the future: We need to define whether these reads should reflect pending writes or whether we should reject reads for locations that were written to.